lib/rules/selector-anb-no-unmatchable/index.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const { parse } = require('css-tree');
+const {
+	aNPlusBNotationPseudoClasses,
+	aNPlusBOfSNotationPseudoClasses,
+} = require('../../reference/selectors');
+const isStandardSyntaxRule = require('../../utils/isStandardSyntaxRule');
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'selector-anb-no-unmatchable';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (pseudoClass) => `Unexpected unmatchable An+B selector "${pseudoClass}"`,
+	rejectedParseError: (selector) => `Cannot parse selector "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/selector-anb-no-unmatchable',
+};
+
+function isUnmatchableNth(/** @type {import('css-tree').AnPlusB} */ nth) {
+	const { a, b } = nth;
+
+	if (a !== null && a !== '0' && a !== '-0') {
+		return false;
+	}
+
+	if (b !== null && b !== '0' && b !== '-0') {
+		return false;
+	}
+
+	return true;
+}
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) {
+				return;
+			}
+
+			ruleNode.selectors.forEach((selector, i) => {
+				try {
+					const cssTreeSelector = parse(selector, { context: 'selector', positions: true });
+
+					checkSelector(cssTreeSelector);
+				} catch (e) {
+					const index = ruleNode.selectors
+						.slice(0, i)
+						.map((sel) => sel.length)
+						.reduce((c, a) => c + a, 0);
+					const endIndex = index + selector.length;
+
+					report({
+						message: messages.rejectedParseError(selector),
+						node: ruleNode,
+						index,
+						endIndex,
```

Comment:
[suggest]

- We may simply use `word` instead of calculating `index` and `endIndex` because a parse error seems to rarely occur.
- How about using `messageArgs`? See #6589

```suggestion
					report({
						message: messages.rejectedParseError,
						messageArgs: [selector],
						node: ruleNode,
						word: selector,
```